### ISSUE TITLE
HasValue::getDefaultValidator

### ISSUE THREAD

**Issue by pekam** [vaadin/flow#6803]
Migrate https://github.com/vaadin/framework/pull/9532 to Flow.

It might be that the `HasValidation` interface (which doesn't exist in V8) is the more appropriate place for this method instead of `HasValue`.

This is needed in order to make Binder work with the internal validation logic built into Flow components. Currently, Binder validation overrides the validation status set by the component validation. Instead, Binder should ask for `getDefaultValidator` from the component and thus take it's internal validation logic into account.

For example, `NumberField` should override `getDefaultValidator` to take `NumberField::setMin` and `NumberField::setMax` into account.

Open question: Should the interface somehow provide means to override the error message of the default validator? In Vaadin 8, `DateField` at least has a specific method for this: `setDateOutOfRangeMessage(String)`.
In V14 we have much more of this kind of component specific server-side validation APIs, and adding a similar error message setter for all of them could bloat the API.
Maybe it's not possible to have this as a (proper) generic API in the interface, and specific method signatures are needed.

**Comment by pleku** [comment]
#### Acceptance Criteria
- [ ] Forward port this the changes from FW 8 to Flow, make sure everything in included
  - Component changes are not part of this
- [ ] Pick the changes to 3.1

**Comment by mshabarov** [comment]
There is already an interface `HasValidator` in the Flow containing `getDefaultValidator()` method and being used by `Binder`. It was added in 2017 according to git history. Components can implement this interface to add internal server-side validator invoked by `Binder`.
@pekam could you please close this ticket, if you agree with this approach?

**Comment by pekam** [comment]
Thanks @mshabarov! I agree. This ticket is based on my misunderstanding. I expected `getDefaultValidator` to be either in `HasValue` (as in V8) or in `HasValidation`. I didn't expect there to be yet another interface `HasValidator` in a different package, not currently implemented by any component (AFAIK).

Anyway, this is good news, as the component updates are not blocked by a Flow feature to be released.
